client/src/redux/sagas/buildSetSaga.js

```diff
@@ -20,6 +20,32 @@ function getBuildSetReqString(commits, buildData) {
   .join('&');
 }
 
+//=>matchShaToBuild matches buildsetid from filteredBS and bsID
+//=>then bsID.buildrequestid is matched with buildData.buildrequestid
+//=>An object is array of object is created with key === sha and
+//=> value === array of objects containing buildData filtered and stored in buildStatus
+
+function matchShaToBuild(buildData, filteredBS, bsID, shas) {
+ return filteredBS.map((obj, index) => {
+  let buildObj = {};
+  let filterObj = [];
+  obj.forEach(val => {
+   filterObj.push(bsID.filter(item => item.buildsetid === val.bsid));
+  });
+  console.log(filterObj);
+  let buildStatus = [];
+  filterObj.forEach(val =>
+   val.forEach(build => {
+    buildStatus.push(
+     buildData.filter(bReqId => bReqId.buildrequestid === build.buildrequestid)
+    );
+   })
+  );
+  buildObj[shas[index]] = buildStatus.flat();
+  return buildObj;
+ });
+}
+
 function* handleBuildsLoad() {
  try {
   const buildSets = yield call(fetchBuildSets);
```

Comment:
Wouldn't that work too?
```suggestion
const commits = yield select(commitSha);
const buildSetsRaw = yield call(fetchBuildSets);

if (buildSetsRaw.length == 0) {
	yield put(setBuildSetsError("Nothing returned"))
	return
}

const buildReqStr = getBuildSetReqString(commits, buildSets)
const buildReqsRaw = yield call(fetchBuildReq, buildReqStr)

const buildsStr = buildStr(buildReqsRaw)
const builds = yield call(fetchBuilds, buildstr)

const buildsBySha = {}

for (let {sha} of commits)
{
	const buildSetIds = buildSetsRaw.filter(bs => bs.sourcestamps[0].revision === commit.sha).map(bs => bs.bsid)
	const buildReqIds = buildReqsRaw.filter(br => buildSerIds.includes(br.bsid)).map(br => br.buildrequestid)
	const builds = buildsRaw.filter(b => buildReqIds.includes(b))

	buildsBySha[sha] = builds;
}

yield put(setBuilds(output))
```